clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoClient.java

```diff
@@ -5,220 +5,85 @@
 import com.datastrato.gravitino.exceptions.NoSuchMetalakeException;
 import com.google.common.base.Preconditions;
-import java.io.Closeable;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * Gravitino Client for interacting with the Gravitino API, allowing the client to list, load,
- * create, and alter Metalakes.
+ * Gravitino Client for an user to interact with the Gravitino API, allowing the client to list,
+ * load, create, and alter Catalog.
  *
  * <p>It uses an underlying {@link RESTClient} to send HTTP requests and receive responses from the
  * API.
  */
-public class GravitinoClient implements SupportsMetalakes, Closeable {
+public class GravitinoClient extends GravitinoClientBase implements SupportsCatalogs {
 
   private static final Logger LOG = LoggerFactory.getLogger(GravitinoClient.class);
 
-  private final RESTClient restClient;
-
-  private static final String API_METALAKES_LIST_PATH = "api/metalakes";
-  private static final String API_METALAKES_IDENTIFIER_PATH = "api/metalakes/";
+  private final GravitinoMetaLake metaLake;
 
   /**
    * Constructs a new GravitinoClient with the given URI, authenticator and AuthDataProvider.
    *
    * @param uri The base URI for the Gravitino API.
+   * @param metalakeName The specified metalake name.
    * @param authDataProvider The provider of the data which is used for authentication.
+   * @throws NoSuchMetalakeException if the metalake with specified name does not exist.
    */
-  private GravitinoClient(String uri, AuthDataProvider authDataProvider) {
-    this.restClient =
-        HTTPClient.builder(Collections.emptyMap())
-            .uri(uri)
-            .withAuthDataProvider(authDataProvider)
-            .build();
+  private GravitinoClient(String uri, String metalakeName, AuthDataProvider authDataProvider) {
+    super(uri, authDataProvider);
+    this.metaLake = loadMetalake(NameIdentifier.of(metalakeName));
   }
 
   /**
-   * Retrieves a list of Metalakes from the Gravitino API.
+   * Get the current metalake object
    *
-   * @return An array of GravitinoMetaLake objects representing the Metalakes.
+   * @return the {@link GravitinoMetaLake} object
+   * @throws NoSuchMetalakeException if the metalake with specified name does not exist.
    */
-  @Override
-  public GravitinoMetaLake[] listMetalakes() {
-    MetalakeListResponse resp =
-        restClient.get(
-            API_METALAKES_LIST_PATH,
-            MetalakeListResponse.class,
-            Collections.emptyMap(),
-            ErrorHandlers.metalakeErrorHandler());
-    resp.validate();
-
-    return Arrays.stream(resp.getMetalakes())
-        .map(o -> DTOConverters.toMetaLake(o, restClient))
-        .toArray(GravitinoMetaLake[]::new);
+  private GravitinoMetaLake getMetaLake() {
```

Comment:
It is better to change to "Metalake", not "MetaLake".

Reply:
It is easy to change. But I see the class "GravitinoMetaLake" has been there for some time, shall I change it as well?

Reply:
Yep, it would be better to change also.

Reply:
Done, please review.